src/storage/query/GetNeighborsProcessor.cpp

```diff
@@ -61,39 +61,71 @@ void GetNeighborsProcessor::doProcess(const cpp2::GetNeighborsRequest& req) {
         }
     }
 
-    auto plan = buildPlan(&resultDataSet_, limit, random);
-    std::unordered_set<PartitionID> failedParts;
-    for (const auto& partEntry : req.get_parts()) {
-        auto partId = partEntry.first;
-        for (const auto& row : partEntry.second) {
-            CHECK_GE(row.values.size(), 1);
-            auto vId = row.values[0].getStr();
-
-            if (!NebulaKeyUtils::isValidVidLen(spaceVidLen_, vId)) {
-                LOG(ERROR) << "Space " << spaceId_ << ", vertex length invalid, "
-                           << " space vid len: " << spaceVidLen_ << ",  vid is " << vId;
-                pushResultCode(nebula::cpp2::ErrorCode::E_INVALID_VID, partId);
-                onFinished();
-                return;
-            }
+    if (!withDst_) {
```

Comment:
If we don't implement this feature by push down the filter. Maybe you could save a dstId in `PlanContext` or `EdgeContext`, and just pass it to `StorageIterator`, the template seems unnecessary.

But I sill prefer by adding a filter.

Reply:
There are pair of <src,dst> for each edge scan, so I think it should passed by the StoragePlan::go method.